### PR TITLE
[stable/spinnaker] Allow service-settings to be extended with kubernetes configuration

### PR DESCRIPTION
**Pull request by darend**
#### What this PR does / why we need it:

This extends the spinnaker service-settings-configmap to support custom kubernetes configuration. For reference https://www.spinnaker.io/reference/halyard/custom/#custom-service-settings

The use case I am trying to support is adding custom annotation to Spinnaker services, such as those required by [kiam](https://github.com/uswitch/kiam). I first tried using the [additionalProfileConfigMaps](https://github.com/helm/charts/blob/e62284c1e5db576530f6ea74dc1705449acc0176/stable/spinnaker/values.yaml#L23-L37) setting but found that while it was mounted into the halyard pod, it was not applied. I then tried this approach which worked.

#### Special notes for your reviewer:

In its current form this PR is more of a POC to get feedback on the approach, as I am not sure if this is the correct way to solve the use case. If it is, Ill expand the service-settings to allow for all services to be configured, update docs, etc

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [ ] Variables are documented in the README.md

### CODE REVIEW

**Comment by kenske** [comment]
It seems like this PR would solve the issues in #12032 #12520 and #12719

File: stable/spinnaker/templates/configmap/service-settings.yaml [hunk 1]
```diff
@@ -31,3 +34,11 @@ data:
...
       serviceType: NodePort
     {{- end }}
+    {{- if .Values.halyard.additionalServiceSettings.deck.kubernetes }}
+{{ toYaml .Values.halyard.additionalServiceSettings.deck.kubernetes | indent 6 }}
+    {{- end }}
+  {{- if .Values.halyard.additionalServiceSettings.front50.kubernetes }}
+  front50.yml: |-
+    kubernetes:
+{{ toYaml .Values.halyard.additionalServiceSettings.front50.kubernetes | indent 6 }}
+  {{- end }}
```
Comment (dwardu89) [inline stable/spinnaker/templates/configmap/service-settings.yaml]: Can you add a new line here please

Comment (darend) [reply stable/spinnaker/templates/configmap/service-settings.yaml]: https://github.com/helm/charts/pull/11631/commits/542ef7f0ca14a215a93b9b6566d66511a0031783

**Comment by darend** [comment]
@paulczar @dwardu89 can you give the `/ok-to-test`?

**Comment by legal90** [comment]
Hi @darend. Thanks for what you;ve done in this PR
Please have a look at latest state of #13049 - it allows to add custom settings for any Spinnaker service in a pretty generic way (not only for `kubernetes` block). What do you think about that?

**Comment by darend** [comment]
> Hi @darend. Thank you for what you've done in this PR!
> Please have a look at latest state of #13049 - it allows to add custom settings for any Spinnaker service in a pretty generic way (not only for kubernetes block). What do you think about it?

Thanks @legal90 , that is a cleaner and more extensible solution. Ill wait for the maintainers to chime in, and can then close this PR

**Comment by darend** [comment]
Replaced by https://github.com/helm/charts/pull/13049

**Pull request closed without merging**